.circleci/config.yml

```diff
@@ -0,0 +1,38 @@
+version: 2.1
+jobs:
+  test:
+    parameters:
+        docker_image:
+          type: string
+          default: cimg/node:current
+    docker:
+      - image: << parameters.docker_image >>
+    steps:
+      - checkout
+      - run:
+          name: Setup
+          command: |
+            rm -rf node_modules package-lock.json
```

Comment:
So that the install is fresh, from scratch. Pretty much all of the ilib libraries and tools run off the latest versions using carat dependencies and should be installed with npm. If a carat dependency stops working for some odd reason, we should put in a fixed dependency specification in the package.json instead of having the package-lock.json save us from ourselves.

In general, I don't like the package-lock.json or the yarn.lock way of doing things. Basically, if I put in the dependencies that I want the latest, it should get me the latest, period. The tool should never ever tell me that it knows better than me what I want.

Yarn even goes so far as to secretly swap out packages I installed locally for debugging with some old cached version. Even when you run `yarn cache clean`, it still does it anyways. I have no clue where it gets it from if the cache is cleared, but somehow, it does. There is some programmer somewhere who thinks they are being clever, but instead, they create headaches and mysteries while debugging. I have had on multiple occasions thought I found a bug, fixed it in code, updated the package, installed it locally, tested it, and it didn't work. Why? Not because I didn't fix the bug, but because yarn refused to install the package I just built. Then, I spend hours trying to figure out why my fix didn't work and what else could be wrong instead, only to find out in the end that my fix was correct all along. That's just big frickin waste of time.

Reply:
Ok, got it.
It's not good to hear the history of `yarn` though. I've never been in a situation like this, but I understand. I would love to see any of your `yarn` problems if they arise again so that I can better understand. Also, `yarn` has a [global cache](https://yarnpkg.com/cli/cache/clean) that you need to clean with a separate command.